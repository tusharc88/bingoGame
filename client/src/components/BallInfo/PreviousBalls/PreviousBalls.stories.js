import React from 'react';
import { storiesOf } from '@storybook/react';
import centered from '@storybook/addon-centered';
import { withKnobs, array, number, text } from '@storybook/addon-knobs';
import PreviousBalls from './index';

storiesOf('BallInfo', module)
  .addDecorator(withKnobs)
  .addDecorator(centered)
  .addWithInfo(
    'PreviousBalls',
    <PreviousBalls
      dimension={number('dimension', 15)}
      numData={array('numData', [1, 2, 3, 4])}
      textVal={text('textVal', 'Previous Balls')}
    />,
    () => (
      <PreviousBalls
        dimension={number('dimension', 15)}
        numData={array('numData', [1, 2, 3, 4])}
        textVal={text('textVal', 'PREVIOUS BALLS')}
      />
    )
  );
