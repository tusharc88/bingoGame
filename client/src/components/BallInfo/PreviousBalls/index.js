import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import BallInfo from '../index';
import Tag from '../../Tag/index';

const PreviousBallsInfoWrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
`;

const PreviousBallsTextWrapper = styled.div`
  font-size: ${props => `${props.dimension} px`}
  width: 95%;
  margin: 10px;
`;

const PreviousBallsNumberDataWrapper = styled.div`
  flex: 1;
  display: flex;
  justify-content: space-around;
`;

const PreviousBalls = ({ dimension, numData, textVal }) => (
  <BallInfo numberData={numData} textData={textVal}>
    {(numberArray, textInfo) => (
      <PreviousBallsInfoWrapper>
        <PreviousBallsTextWrapper dimension={dimension}>
          {textInfo}
        </PreviousBallsTextWrapper>
        <PreviousBallsNumberDataWrapper>
          {numberArray.map(
            num =>
              num && (
                <Tag dimension={dimension} key={num}>
                  {num}
                </Tag>
              )
          )}
        </PreviousBallsNumberDataWrapper>
      </PreviousBallsInfoWrapper>
    )}
  </BallInfo>
);

PreviousBalls.propTypes = {
  dimension: PropTypes.number.isRequired,
  numData: PropTypes.arrayOf(PropTypes.number).isRequired,
  textVal: PropTypes.string.isRequired,
};

export default PreviousBalls;
