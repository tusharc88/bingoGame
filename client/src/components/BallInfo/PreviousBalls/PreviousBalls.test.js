import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import PreviousBalls from './index';

test('Renders with default props', () => {
  const tree = renderer
    .create(
      <PreviousBalls
        dimension={15}
        numData={[1, 2, 3, 4]}
        textVal="PREVIOUS BALLS"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
