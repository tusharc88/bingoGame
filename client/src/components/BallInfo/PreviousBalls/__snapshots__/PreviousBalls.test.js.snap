// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders with default props 1`] = `
.c0 {
  width: 100%;
}

.c4 {
  height: 30px;
  width: 45px;
  background-color: #edf0f3;
  text-align: center;
  vertical-align: middle;
  font-size: 15px;
  line-height: 30px;
  border-radius: 5px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c2 {
  font-size: 15 px;
  width: 95%;
  margin: 10px;
}

.c3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      PREVIOUS BALLS
    </div>
    <div
      className="c3"
    >
      <div
        className="c4"
      >
        1
      </div>
      <div
        className="c4"
      >
        2
      </div>
      <div
        className="c4"
      >
        3
      </div>
      <div
        className="c4"
      >
        4
      </div>
    </div>
  </div>
</div>
`;
